wom/src/main/scala/wdl/WdlNamespace.scala

```diff
@@ -455,13 +455,18 @@ object WdlNamespace {
     invalidVariableReferences ++ typeMismatches
   }
 
-  private def lookupType(from: Scope)(n: String): WdlType = {
+  def lookupType(from: Scope)(n: String): WdlType = {
     val resolved = from.resolveVariable(n)
     resolved match {
       case Some(d: DeclarationInterface) => d.relativeWdlType(from)
       case Some(c: WdlCall) => WdlCallOutputsObjectType(c)
       case Some(s: Scatter) => s.collection.evaluateType(lookupType(s), new WdlStandardLibraryFunctionsType, Option(from)) match {
-        case Success(a: WdlArrayType) => a.memberType
+        case Success(a: WdlArrayType) =>
+          // Collection is an array
+          a.memberType
+        case Success(WdlMapType(kType,vType)) =>
```

Comment:
This could be covered by the custom unapply of `WdlArrayType`, eg this one-liner should work:
```
case Success(WdlArrayType(arrayType)) => arrayType
```

Reply:
ToL (i.e. unrelated to this PR) I would not have expected that. Perhaps we should have some "array-like" trait that the array and map types can share that would make this less surprising.